server/src/main/scala/com/cloudera/livy/server/AccessFilter.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Cloudera, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Cloudera, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.livy.server
+
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.cloudera.livy.LivyConf
+
+class AccessFilter extends Filter {
+
+  val allowedUsers = scala.collection.mutable.Set[String]()
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    val users = filterConfig.getInitParameter(LivyConf.ACCESS_CONTROL_USERS.key)
```

Comment:
I'd suggest doing something similar to the `superusers` method in `LivyConf`. Then you can reuse some of the code for parsing lists too, and avoid the mutable set (not a big deal but I prefer to avoid mutable types unless really needed).